common/logger/src/diem_logger.rs

```diff
@@ -258,11 +304,11 @@ impl DiemFilter {
     }
 }
 
-///
 pub struct DiemLogger {
     sender: Option<SyncSender<LoggerServiceEvent>>,
     printer: Option<Box<dyn Writer>>,
     filter: RwLock<DiemFilter>,
+    pub(crate) formatter: Box<dyn Fn(&LogEntry) -> Result<String, fmt::Error> + Send + Sync>,
```

Comment:
I don't think this needs to be marked as `pub(crate)`

Reply:
Also there shouldn't be a need for this to be a Box<dyn Fn> instead you should change all of these to be
`fn(&LogEntry) -> Result<String, fmt::Error>`

Notice the lowercase `f` in `fn`, not `Fn`.

Reply:
I just pushed a commit showing this

Reply:
thanks!